### PR TITLE
 Improve calendar get events with query param support

### PR DESCRIPTION
**Pull request by Chuhaa**
## Purpose
Add optional parameters in get events.

## Goal
Query events in get events response.

### CODE REVIEW

File: calendar/records.bal [hunk 1]
```diff
@@ -582,3 +582,53 @@ public type CalendarResponse record {
...
+# + q - Free text search terms to find events that match these terms in any field, except for extended properties
+# + sharedExtendedProperty - Extended shared properties constraint specified as propertyName=value
+# + showDeleted - Whether to include deleted events (with status equals "cancelled") in the result
+# + showHiddenInvitations - Whether to include hidden invitations in the result
+# + singleEvents - Whether to expand recurring events into instances and only return single one-off events
+# + timeMax - Upper bound (exclusive) for an event's start time to filter by. Must be an RFC3339 timestamp
+# + timeMin - Lower bound (exclusive) for an event's end time to filter by. Must be an RFC3339 timestamp
+# + timeZone - Time zone used in the response
+# + updatedMin - Lower bound for an event's last modification time (as a RFC3339 timestamp) to filter by
+public type ListEventsToAccess record {
```
Comment (abeykoon) [inline calendar/records.bal:601]: Shall we raname this record to `EventFilterCriteria` as it contains params related to calender event filtering?

File: calendar/endpoints.bal [hunk 2]
```diff
@@ -166,12 +166,14 @@ public client class Client {
     # 
     # + calendarId - Calendar id
     # + userAccount - The email address of the user for requesting delegated access in service account
+    # + optional - Record that contains optional parameters
     # + return - Event stream on success, else an error
     @display {label: "Get Events"}
     remote isolated function getEvents(@display {label: "Calendar Id"} string calendarId,
+                                        @display {label: "Events to Access"} ListEventsToAccess? optional = (),
```
Comment (abeykoon) [inline calendar/endpoints.bal:173]: `Filtering Criteria`

File: calendar/records.bal (hunk 1 above)
Comment (Chuhaa) [reply calendar/records.bal]: Fixed in https://github.com/ballerina-platform/module-ballerinax-googleapis.calendar/commit/67761b55d161fbf36993eadfc4b8ee43652cc3c9

File: calendar/endpoints.bal (hunk 2 above)
Comment (Chuhaa) [reply calendar/endpoints.bal]: Fixed in https://github.com/ballerina-platform/module-ballerinax-googleapis.calendar/commit/67761b55d161fbf36993eadfc4b8ee43652cc3c9

**Review by abeykoon — APPROVED** [review]

**Pull request merged**